src/p2p/net_node.inl

```diff
@@ -71,6 +71,22 @@
 
 #define MIN_WANTED_SEED_NODES 12
 
+namespace
+{
+
+boost::asio::ip::address_v4 make_address_v4_from_v6(boost::asio::ip::v4_mapped_t m, const boost::asio::ip::address_v6& a)
+{
+  const auto &bytes = a.to_bytes();
```

Comment:
The boost implementation has an additional check:
```c++
if (!v6_addr.is_v4_mapped())
  {
    bad_address_cast ex;
    boost::asio::detail::throw_exception(ex);
  }
```
Might be worth doing, and if so I would move to a cpp file.

Reply:
It's only called after that check. Not really worth it I think.